### ISSUE TITLE
Automatic differentiation to calculate derivatives

### ISSUE THREAD

**Issue by leila-pujal** [theochem/gbasis#134]
**Is there a specific integral/formula that you would like implemented?**
Current Gbasis code performs arbitrary order derivatives by using specific formula involving Hermite polynomials. Based on later talks, it would be interesting to prototype an test a version of Gbasis that could use automatic differentiation. This would give access to nuclear derivatives as well, which is a missing feature. 

**Is there a change to the code or algorithm you would like to see?**
Right now to compute a specific order derivative you would use funct [evaluate_deriv_reduced_density_matrix](https://github.com/theochem/gbasis/blob/e367156752681496cea2e72458ee7424c5058947/gbasis/evals/density.py#L8C73-L8C73) and specify the orders for first and second grid points. This ultimately calls [_eval_deriv_contraction](https://github.com/theochem/gbasis/blob/e367156752681496cea2e72458ee7424c5058947/gbasis/evals/_deriv.py#L11). The goal would be to use Automatic differentiation to avoid the explicit use of this function.

**Comment by RichRick1** [comment]
@leila-pujal I think the tentative plan is to use the autograd library for these purposes. Let me know if you think other libraries will be more suitable for this.

**Comment by leila-pujal** [comment]
Yes, I think we agreed on that one. I know about Automatic differentiation but I've never used it in code. I think you are the most experienced in this field, so if you think autograd is the best one or simply it is the one you know best then let's use it. If you start working on this and have any doubts regarding Gbasis code, please feel free to write a message on this issue.

**Comment by PaulWAyers** [comment]
This is a good idea, and I'd talked about it with @FarnazH before. We should prototype though to see if it is faster or slower than the analytic formulas. For nuclear derivatives, it is definitely a good idea.

**Comment by RichRick1** [comment]
@msricher @PaulWAyers you told me that libcint doesn't allow us to evaluate derivatives higher than 2. Maybe this will help https://github.com/autodiff/autodiff
It seems like this library is pure analog of autograd, but in c++, so we can gain a lot of speed potentially
